### PR TITLE
Implemented 13.3.5.  Caller Identification

### PR DESCRIPTION
**Pull request by dudochkin-victor**
Implemented 13.3.5.  Caller Identification

### CODE REVIEW

**Review by mourad — CHANGES_REQUESTED** [review]
Looks good, some minor comments below.

File: examples/rpc-with-options/rpc-server/main.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
+package main
+
+import (
+	"log"
+	"net/http"
+	"time"
+
+	"gopkg.in/jcelliott/turnpike.v2"
+	//"github.com/dudochkin-victor/turnpike"
```
Comment (mourad) [inline examples/rpc-with-options/rpc-server/main.go]: Here as well.

File: examples/rpc-with-options/rpc-client/main.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+import (
+	"bufio"
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+
+	"gopkg.in/jcelliott/turnpike.v2"
+	//"github.com/dudochkin-victor/turnpike"
```
Comment (mourad) [inline examples/rpc-with-options/rpc-client/main.go]: Remove commented import.

File: examples/rpc-with-options/README.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+2. Run the client with `go run rpc-client/main.go`
+3. ???
+4. PROFIT!!!
+
+Now the rpc method knows caller ;)
+
+Check the import which you use.
+Choose proper:
+	"gopkg.in/jcelliott/turnpike.v2"
+	//"github.com/dudochkin-victor/turnpike"
```
Comment (mourad) [inline examples/rpc-with-options/README.md]: Seem that you could remove the last 4 lines.

File: dealer.go [hunk 4]
```diff
@@ -124,19 +124,29 @@ func (d *defaultDealer) Call(caller *Session, msg *Call) {
 			})
 		} else {
 			// everything checks out, make the invocation request
-			// TODO: make the Request ID specific to the caller
+			// TODO: make the Request ID specific to the caller <<< wrong solution
```
Comment (mourad) [inline dealer.go]: Could you elaborate?

**Review by mourad — CHANGES_REQUESTED** [review]

**Comment by dudochkin-victor** [comment]
about imports: i just need a working version of code. (so keep the coments:))) )
about "TODO: make the Request ID specific to the caller". The dealer create the own id for request (not for caller). i do not think that this field should be related with caller. check the spec. 
my email: dudochkin.victor@gmail.com.
skype: dudochkin-victor

**Comment by dudochkin-victor** [comment]
there is one thing is that the method may require the opening session. so it must be solved on the dealer side

**Comment by dudochkin-victor** [comment]
and I still have not solved this problem

**Comment by dudochkin-victor** [comment]
i will remove comments when u able to accept the code. few minutes to create new request

**Review by mourad — APPROVED** [review]

**Pull request merged**